### PR TITLE
Adds Security UI Considerations.

### PR DESCRIPTION
**Pull request by mfoltzgoogle**
This PR addresses Issue #118: UI guidelines for pairing and trusted/untrusted data.  
Action from the Berlin F2F: https://www.w3.org/2019/05/23-webscreens-minutes.html#x21

It adds a Security UI Considerations section with general guidelines on how metadata from agents should be presented to the user, and on PSK display and input. 

It also defines a "suspicious agent" as one that meets a criterion for a potential MITM attack.  The details on how to display a suspicious agent are moved to Security UI Considerations.

It also moves the description of how Instance Names should be handled into Security UI Considerations.

Finally it clarifies that any agents advertising a public key FP collision should be treated as suspicious (trusted or not) since we won't have any way to tell them apart.

***
<a href="https://pr-preview.s3.amazonaws.com/webscreens/openscreenprotocol/pull/197.html" title="Last updated on Aug 30, 2019, 3:28 AM UTC (78eebc7)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webscreens/openscreenprotocol/197/8e25af6...78eebc7.html" title="Last updated on Aug 30, 2019, 3:28 AM UTC (78eebc7)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -1982,6 +1960,52 @@ href="https://en.wikipedia.org/wiki/Sandbox_(computer_security)">sandboxing</a>
 to prevent vulnerabilities from gaining access to user data or leading to
 persistent exploits.
 
+User Interface Considerations {#security-ui}
+-----------------------------
+
+This specification does make any specific requirements of the security relevant
```
Comment (anssiko) [inline index.bs]: does not

File: index.bs [hunk 2]
```diff
@@ -1982,6 +1960,52 @@ href="https://en.wikipedia.org/wiki/Sandbox_(computer_security)">sandboxing</a>
...
+user interfaces of Open Screen agents.  However there are important
+considerations when designing these user interfaces, as PSK based authentication
+requires users to make informed decisions about which agents to trust.
+
+1. Before an agent has authenticated another device, the agent should make it
+    clear that any `agent-info` or other data from that device has not been
+    verified by authentication.  (See below for how this applies to DNS-SD
+    Instance Names.)
+1. An [=suspicious agent=] should be displayed differently from trusted
```
Comment (chrisn) [inline index.bs]: Nit: "An" should be "A"

File: index.bs [hunk 3]
```diff
@@ -1982,6 +1960,52 @@ href="https://en.wikipedia.org/wiki/Sandbox_(computer_security)">sandboxing</a>
...
+1. The user interface to input a PSK during authentication should be done in
+    trusted UI and be difficult to spoof.
+1. The user should be required to take action to input the PSK, to prevent the
+    user from blindly clicking through this step.
+1. The user interfaces to render and input a PSK should meet accessibility
+    guidelines.
+
+### Instance and Display Names  ### {#instance-names}
+
+Because DNS-SD [=Instance Names=] are the the primary information that the user
```
Comment (chrisn) [inline index.bs]: Nit: "the the"

**Comment by mfoltzgoogle** [comment]
Thanks - perhaps @nigelcearnshaw can provide feedback on what was merged.

File: index.bs [hunk 4]
```diff
@@ -1987,6 +1965,52 @@ href="https://en.wikipedia.org/wiki/Sandbox_(computer_security)">sandboxing</a>
 to prevent vulnerabilities from gaining access to user data or leading to
 persistent exploits.
 
+User Interface Considerations {#security-ui}
+-----------------------------
+
+This specification does not make any specific requirements of the security
+relevant user interfaces of Open Screen agents.  However there are important
+considerations when designing these user interfaces, as PSK based authentication
```
[the remark this answers is not in the document]
Comment (mfoltzgoogle) [reply index.bs]: Fixed.

**Pull request merged**